src/main/java/homer/common/limit/Limit.java

```diff
@@ -30,7 +33,37 @@ public static int limit(final int input, final int lowerLimit, final int upperLi
      * @return The input limited between the lower and upper limits.
      */
     public static double limit(final double input, final double lowerLimit, final double upperLimit) {
+        checkLimitsOrder(lowerLimit, upperLimit);
         return Math.max(lowerLimit, Math.min(input, upperLimit));
     }
 
+    /**
+     * Limits the input between lower and upper limits.
+     * 
+     * @param <S>        The comparable numeric type.
+     * @param input      The input to limit.
+     * @param lowerLimit The lower limit.
+     * @param upperLimit The upper limit.
+     * @return The input limited between the lower and upper limits.
+     */
+    public static <S extends Number & Comparable<S>> S limit(final S input, final S lowerLimit, final S upperLimit) {
```

Comment:
@michelececcacci The bound on `Number` could be removed and it would allow limiting practically any comparable type, including `Temperature` I guess, I will do that tomorrow.

Reply:
that's fine for me, will probably require a pull if it's being used somewhere else
@aleemont1

Reply:
I will do this in a separate pr